### PR TITLE
resolve comments from #11650

### PR DESCRIPTION
**Pull request by JasonLiuZhuoCheng**
#### Change overview
Fix pr comments in https://github.com/project-chip/connectedhomeip/pull/11650

#### Testing
manually tested.

### CODE REVIEW

File: src/controller/java/src/chip/clusterinfo/InteractionInfo.java [hunk 1]
```diff
@@ -5,16 +5,16 @@
 import java.util.function.Supplier;
 
 /**
- * CommandInfo has a functional interface to invoke arbitrary commands based on cluster, callback
- * and a map of arguments, a Supplier that provides {@link DelegatedClusterCallback}, and maps the
- * parameter and commandParametersInfo.
+ * InteractionInfo has a functional interface to invoke arbitrary commands based on cluster,
```
Comment (austinh0) [inline src/controller/java/src/chip/clusterinfo/InteractionInfo.java:8]:
```suggestion
 * InteractionInfo has a functional interface to invoke arbitrary interactions based on cluster,
```

File: src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/clusterinteraction/ClusterDetailFragment.kt [hunk 2]
```diff
@@ -182,37 +181,55 @@ class ClusterDetailFragment : Fragment() {
...
+    }
+  }
+
+  private fun createPrimitiveReadAttributeView(
+    response: Any,
+    inflater: LayoutInflater,
+    callbackList: LinearLayout,
+    variableNameType: CommandResponseInfo
+  ) {
+    val callback =
```
Comment (austinh0) [inline src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/clusterinteraction/ClusterDetailFragment.kt:197]: callbackItem?

File: src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/clusterinteraction/ClusterDetailFragment.kt [hunk 3]
```diff
@@ -182,37 +181,55 @@ class ClusterDetailFragment : Fragment() {
...
-        }
+        createListReadAttributeView(response, inflater, callbackList, variableNameType)
       } else {
-        val callback =
+        createPrimitiveReadAttributeView(response, inflater, callbackList, variableNameType)
+      }
+    }
+  }
+
+  private fun createPrimitiveReadAttributeView(
```
Comment (austinh0) [inline src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/clusterinteraction/ClusterDetailFragment.kt:191]: This method covers single `String` and `byte[]` response types, which are not primitives. Maybe `createBasicReadAttributeView`?

**Review by austinh0 — APPROVED** [review]
Merge conflict in ClusterInfoMapping, please regenerate.

File: src/controller/java/BUILD.gn [hunk 4]
```diff
@@ -69,10 +69,14 @@ android_library("java") {
   sources = [
     "src/chip/clusterinfo/ClusterCommandCallback.java",
     "src/chip/clusterinfo/ClusterInfo.java",
-    "src/chip/clusterinfo/CommandInfo.java",
+<<<<<<< HEAD
+=======
+    "src/chip/clusterinfo/InteractionInfo.java",
+>>>>>>> df6a8040a4ba09bfde26c36c14f01433f78541a0
```
Comment (electrocucaracha) [inline src/controller/java/BUILD.gn:75]: This is a merge conflict which needs to be resolved first

File: src/controller/java/templates/ClusterInfo-java.zapt [hunk 5]
```diff
@@ -250,13 +250,20 @@ public class ClusterInfoMapping {
  public Map<String, ClusterInfo> getClusterMap() {
     Map<String, ClusterInfo> clusterMap = new HashMap<>();
     getCommandMap(clusterMap);
-    getReadAttributeMap(clusterMap);
+    Map<String, Map<String, InteractionInfo>> readAttributeMap = getReadAttributeMap();
+    combineCommand(clusterMap, readAttributeMap);
```
Comment (austinh0) [inline src/controller/java/templates/ClusterInfo-java.zapt:254]: Like we discussed earlier, could we have something like this?
```
Map<String, ClusterInfo> clusterMap = initializeClusterMap();
Map<String, Map<String, InteractionInfo>> commandMap = getCommandMap(clusterMap);
combineCommand(clusterMap, commandMap);
Map<String, Map<String, InteractionInfo>> readAttributeMap = getReadAttributeMap();
combineCommand(clusterMap, readAttributeMap);
```
We want `getCommandMap` and `readAttributeMap` to be consistent.

**Comment by andy31415** [comment]
Fast track: platform specific changes, PR has been up for cross timezone review, PR approved by platform owner.

**Pull request merged**